test/marshaling_test.ts

```diff
@@ -72,6 +83,52 @@ describe('Marshaling', () => {
     assert.deepEqual(transform(new SomeClass('hi')), {message: 'hi'});
   });
 
+  it('can write values readable on node 14 from node 18', async () => {
```

Comment:
FYI, I've added a test case for the node18 backwards compatibility by hard-coding serialized output here. I manually ran just this one test case on node18 and it passed, showing that for at least these cases it's harmless to swap the wire version. I copied these test cases from above, but happy to add more if anything else seems worthwhile